Makefile

```diff
@@ -7,36 +7,54 @@
 # Variables
 #--------------------------------------------------
 PYTHON_FILES?=$$(find biomage -name '*.py')
-INSTALL_PATH=/usr/local/bin
-MKFILE_PATH := $(abspath $(lastword $(MAKEFILE_LIST)))
-MKFILE_DIR := $(dir $(MKFILE_PATH))
+ENTRY_POINT=/usr/bin/biomage
+OLD_ENTRY_POINT=/usr/local/bin/biomage
+
 # should point to biomage-utils root folder to link ./biomage python module & ./venv
 #--------------------------------------------------
 # Targets
 #--------------------------------------------------
 install: clean ## Creates venv, and adds biomage as system command
-	@echo "Creating virtual env and installing dependencies..."
+	@echo "==> Building package..."
+	@python3 setup.py sdist
+	@echo "    [✓]"
+
+	@echo "==> Creating virtual environment..."
 	@python3 -m venv venv
-	@venv/bin/pip3 install --upgrade pip
-	@venv/bin/pip3 install -r requirements.txt
-	@echo "    [✓]\n"
-	@echo "Installing biomage into ${INSTALL_PATH}"
-	@printf '#!/bin/bash\n$(MKFILE_DIR)venv/bin/python3 $(MKFILE_DIR)biomage $$@\n' > ${INSTALL_PATH}/biomage
-	@chmod +x /usr/local/bin/biomage
-	@echo "    [✓]\n"
-fmt: ## Formats python files
+	@echo "    [✓]"
+
+	@echo "==> Installing utility and dependencies..."
+	@venv/bin/pip install --upgrade pip
+	@venv/bin/pip install .
+	@sudo ln -sf '$(CURDIR)/venv/bin/biomage' $(ENTRY_POINT)
+	@echo "    [✓]"
+
+uninstall: clean ## Uninstalls utility and destroys venv
+	@echo "==> Uninstalling utility and dependencies..."
+	@venv/bin/pip uninstall -y biomage-utils
+	@rm -rf venv/
+	@sudo rm -f $(ENTRY_POINT) $(OLD_ENTRY_POINT)
+	@echo "    [✓]"
+
+develop: ## Installs development dependencies
+	@echo "==> Installing development dependencies..."
+	@venv/bin/pip install -r dev-requirements.txt --quiet
+	@echo "    [✓]"
+
+fmt: develop ## Formats python files
 	@echo "==> Formatting files..."
-	@black $(PYTHON_FILES)
-	@echo ""
-check: ## Checks code for linting/construct errors
+	@venv/bin/black $(PYTHON_FILES)
+	@echo "    [✓]"
+
+check: develop ## Checks code for linting/construct errors
 	@echo "==> Checking if files are well formatted..."
-	@flake8 $(PYTHON_FILES)
-	@echo "    [✓]\n"
```

Comment:
why remove the newline?

Reply:
Newlines are printed literally on Linux, so I thought this was an error. I can fix it by using `-e`, if that works on mac.

Reply:
Fixed just by adding a blank `echo` under each tick.